### ISSUE TITLE
Branch for Sonic 2 betas ?

### ISSUE THREAD

**Issue by GabrielRavier** [sonicretro/s2disasm#11]
Should we do a branch for the Sonic 2 protos ?

Because though that would nice. For many hackers.

**Comment by Clownacy** [comment]
There are [already disassemblies](http://info.sonicretro.org/Disassemblies#Sonic_2_Nick_Arcade_prototype) for many of the betas, just not on this repo. The problem is the process of merging them. It's basically impossible without starting over from scratch (and depending on which beta we're talking about, even _that_ may be completely unfeasible).

If we're going to incorporate other disassemblies, I think it's only right to match them to the current disassembly's style. Makes it easier to spot the differences, you know? Not to mention, it helps with compatibility, and avoids the fragmentation that comes with every disassembly having a different style.

Problem is, overhauling a disassembly to closely resemble this one is a crazy task. I remember just overhauling Sonic 1's sound driver from Hivebrain to Git-style being an absolute nightmare of tedium.

That's why, when I came to add REV00 and REV02 disassemblies to this, I started from scratch, using a binary diff on two ROMs, and gradually editing the existing REV01 disassembly until it matched. Even then, this was only feasible because the three revisions are very similar (even if REV02 had me manually adjusting [thousands of different bits of machine-generated linker data](https://github.com/sonicretro/s2disasm/commit/d2db0e23d6cd455c3e13879362ccfd36fea7c241)).

Imagine the amount of work needed to diff Sonic 2 Final from Sonic 2 Nick Arcade? That thing still used Sonic 1's driver: imagine how much else is just from Sonic 1. REV02 pales in comparision to that, and that thing took what felt like weeks of just changing short branches to long jumps, removing JmpTos, and padding dc.b instructions.

I've even been putting off making a 'Knuckles in Sonic 2' branch, because of how much work it would take. Not to toot my own horn, but looking at the commit history, I feel like I'm one of the only people that work on this disassembly anymore, and I'm pretty sure if I were to take on something that makes REV02 look tame, it would probably kill me.

**Comment by GabrielRavier** [comment]
Randomly stumbled upon this 6 years later - I guess you might as well just close this as a WONTFIX if it'll never happen.

**Comment by Clownacy** [comment]
> Randomly stumbled upon this 6 years later - I guess you might as well just close this as a WONTFIX if it'll never happen.

WONTFIX, eh? After nearly 7 years, I've finally added [a branch for Beta 8](https://github.com/sonicretro/s2disasm/tree/beta8)! The next time that I have the motivation, I can chip-away away at the earlier betas too.

**Comment by SONIC123CDMANIA** [comment]
> The next time that I have the motivation, I can chip-away away at the earlier betas too.
The easiest ones to do are the other betas (4-7, I don't know about 6B, though...). Then you could go in a "backwards chronological" order (like using the Beta 8 branch as a basis for a Beta 7 branch) back to Nick Arcade (Whether Sonic 1 (and hence, its prototype) should be included after Nick Arcade is included is a whole other topic).

**Comment by AlexField442** [comment]
I don't think we'd go back _that_ far. At most, I'd see going back to Beta 5, as the rest already have disassemblies available for them.

**Comment by SONIC123CDMANIA** [comment]
> I don't think we'd go back _that_ far. At most, I'd see going back to Beta 5, as the rest already have disassemblies available for them.

Yeah, but are they dissasemblies by Sonic Retro? No they are not. Plus, I'm pretty sure there are protos that DON'T have dissasemblies.

> Beta 8 was done as it is, barring some last minute bug fixes and new demo recordings, effectively the final game, so it wouldn't have been too difficult to do a disassembly comparison. _Nick Arcade_ on the otherhand, is dated around four months from the final (May 1992 vs. September 1992), and is in practice a crazily hacked up _Sonic 1_ with leftovers everywhere.

No, you didn't understand what I said. I said to do it in a "backwards chronological order", meaning it COULD be possible to do Nick Arcade. Like this; Final (literally this disassembly), Beta 8 (branch already made), Beta 7, Beta 6 (And MAYBE 6B? It IS a real prototype from the same day, but whether it should be included is up to everyone else), Beta 5, Beta 4, CENSOR, 9/14, 8/21, Simon Wai, Nick Arcade.
